specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/codeContainer.json

```diff
@@ -0,0 +1,378 @@
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/APIVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/CodeContainerNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/AssetContainerResource"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "machineLearningServices.json#/definitions/MachineLearningServiceError"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "CodeContainer"
+        ],
+        "x-ms-examples": {
+          "Create or Update CodeContainer": {
+            "$ref": "./examples/createOrUpdateCodeContainer.json"
+          }
+        },
+        "description": "Create or update code container.",
+        "operationId": "Code_CreateOrUpdate",
+        "parameters": [
+          {
+            "$ref": "machineLearningServices.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/APIVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/CodeContainerNameParameter"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Model container entity to create or update.",
+            "schema": {
+              "$ref": "#/definitions/CodeVersionResourceRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/AssetContainerResource"
+            }
+          },
+          "201": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/AssetContainerResource"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "machineLearningServices.json#/definitions/MachineLearningServiceError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "CodeContainerNameParameter": {
+      "in": "path",
+      "name": "name",
+      "type": "string",
+      "description": "Name for CodeVersion.",
+      "required": true,
+      "x-ms-parameter-location": "method"
+    }
+  },
+  "definitions": {
+    "AssetContainer": {
+      "type": "object",
+      "properties": {
+        "name": {
```

Comment:
name is a top level field in AssetContainerResource.  It shouldn't be duplicated here.  If this is some sort of other display name please change the property name to make it more clear

Reply:
should these sort of duplicates show up in the top-most level field (ie AssetContainerResource) or the bottom-most level field (ie AssetContainer)?